app/src/main/java/org/xbmc/kore/ui/RemoteActivity.java

```diff
@@ -524,9 +529,15 @@ private Uri getYouTubeUri(String extraText) {
     private String getVideoId(Uri playuri) {
         if (playuri.getHost().endsWith("youtube.com") || playuri.getHost().endsWith("youtu.be") || playuri.getHost().endsWith("vimeo.com")) {
             // We'll need to get the v= parameter from the URL
-            final Pattern pattern =
-                    Pattern.compile("(?:https?:\\/\\/)?(?:www\\.|m\\.)?(?:youtu(?:\\.be\\/|be\\.com\\/watch\\?v=)|vimeo\\.com\\/)([\\w-]+)",
-                            Pattern.CASE_INSENSITIVE);
+            final Pattern pattern;
+            if (playuri.getHost().endsWith("vimeo.com")) {
+                pattern = Pattern.compile("^(?:https?:\\/\\/)?(?:www\\.|player\\.)?vimeo\\.com\\/(?:.*\\/)?(\\d+)(?:\\?.*)?$",
+                        Pattern.CASE_INSENSITIVE);
+            }
+            else {
+                pattern = Pattern.compile("^(?:https?:\\/\\/)?(?:www\\.|m\\.)?youtu(?:\\.be|be\\.com)\\/watch\\?(?:.*&)?v=([\\w-]+)(?:&.*)?$",
+                        Pattern.CASE_INSENSITIVE);
+            }
```

Comment:
This breaks the handling of Youtube URIs shared from the youtube app.

These URIs are of the form http://youtu.be/videoid

Didn't test sharing from the website.